From e790e2489535538ac0a7d1486ca6bb1f002c7700 Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <maarten.lankhorst@canonical.com>
Date: Tue, 16 Jul 2013 10:18:38 +0200
Subject: [PATCH] osmesa: link against static libglapi library too to get the
 gl exports

This should fix missing symbols in a osmesa built against shared glapi
osmesa build. All opengl exports were missing that are defined in the
static glapi, so link against both to fix this.

This is a candidate for the stable series.

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=47824
Signed-off-by: Maarten Lankhorst <maarten.lankhorst@canonical.com>
(cherry picked from commit b20b2b6dc8b72fc099afb9690f61596e1d0b0902)
---
 src/mesa/drivers/osmesa/Makefile.am | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/mesa/drivers/osmesa/Makefile.am b/src/mesa/drivers/osmesa/Makefile.am
index 2503401..4bde617 100644
--- a/src/mesa/drivers/osmesa/Makefile.am
+++ b/src/mesa/drivers/osmesa/Makefile.am
@@ -40,11 +40,11 @@ lib@OSMESA_LIB@_la_SOURCES = osmesa.c
 
 lib@OSMESA_LIB@_la_LDFLAGS = -module -version-number @OSMESA_VERSION@ -no-undefined
 
-if HAVE_SHARED_GLAPI
-GLAPI_LIB = $(top_builddir)/src/mapi/shared-glapi/libglapi.la
-else
 GLAPI_LIB = $(top_builddir)/src/mapi/glapi/libglapi.la
+if HAVE_SHARED_GLAPI
+GLAPI_LIB += $(top_builddir)/src/mapi/shared-glapi/libglapi.la
 endif
+
 lib@OSMESA_LIB@_la_LIBADD = \
 	$(top_builddir)/src/mesa/libmesa.la \
 	$(GLAPI_LIB) \
-- 
1.8.3.2

